static/js/index.js

```diff
@@ -157,90 +142,73 @@ async function initializeIndexAutocomplete() {
-                        return !option.startsWith(prefix);
-                    });
-                    indexValues = filteredIndexValues;
-                    $(this).autocomplete("option", "source", filteredIndexValues);
-                }
             } else {
+                var matcher = new RegExp( typedValue.replace("*",".*") ,"gi" );
+                function hasMatchingString(arr, regex) {
+                    return arr.some((element) => regex.test(element.toLowerCase()));
+                }
+                const matchesIndex =  hasMatchingString(indexValues, matcher);
+                // If the typed value matches any index, add the option to index list
+                if (matchesIndex) {
+                    indexValues.unshift(typedValue);
+                    addSelectedIndex(typedValue);
+                    if (!selectedSearchIndex.split(',').includes(typedValue)) {
+                        selectedSearchIndex += selectedSearchIndex ? ',' + typedValue : typedValue;
+                    }
+                }
+                $(this).autocomplete("option", "source", indexValues);
                 $(this).val('');
                 this.style.width = '5px';
             }
-            if ($(this).autocomplete('widget').is(':visible')) {
-                $(this).autocomplete('close');
-            } else {
-                $(this).autocomplete('search', '');
-            }
+            $(this).autocomplete('close');
+            $(this).blur();
         }
-    }); 
+    });
 }
 
 // Remove selected index from container when remove icon is clicked
 $(".index-container").on("click", ".remove-icon", function(e) {
-    if ($('.index-container .selected-index').length === 1) {
-        return; // If there's only one tag left, do not remove it
-    }
-
     const removedValue = $(this).parent().contents().filter(function() {
         return this.nodeType === 3;
     }).text().trim();
+
+
+    if ($('.index-container .selected-index').length === 1) {
+       // If there's only one tag left, add * as selected index
+       if(removedValue === "*"){
+            return;
+       }else{
+        addSelectedIndex("*")
+        selectedSearchIndex += selectedSearchIndex ? ',' + "*" : "*";
+       }
+    }
+    
     $(this).parent().remove();
 
-    // If the removed value is a wildcard option
-    if (removedValue.endsWith('*')) {
-        const prefix = removedValue.slice(0, -1); // Remove the '*'
-        
-        // Remove wildcard options from indexValues
-        indexValues = indexValues.filter(function(option) {
-            return !option.startsWith(prefix);
-        });
-        
-        // Add back only the non-wildcard options that match the prefix
-        const matchingOptions = originalIndexValues.filter(function(option) {
-            return option.startsWith(prefix) && !selectedSearchIndex.includes(option);
-        });
-        
-        indexValues.push(...matchingOptions);
+    if (!removedValue.endsWith('*')) {        
```

Comment:
Here, we can make it as 
`if (!removedValue.includes('*'))`  , so that when we remove the wildcard, it does not get added to the options."